event.go

```diff
@@ -18,6 +18,11 @@ var eventPool = &sync.Pool{
 	},
 }
 
+// ErrorMarshalFunc allows customization of global error marshaling
+var ErrorMarshalFunc = func (err error) interface{} {
```

Comment:
You could have a nil default and skip the type assertion in the default state. It might improve performance when people not using this feature (to be tested though).

Reply:
I'll write a benchmark and post results.

Reply:
So here's the benchmark https://gist.github.com/DusanKasan/6eb7ef0b4e784bbd26a32df0073ec9cc. 

Results on my machine:
```
BenchmarkCalling-4                      20000000                66.7 ns/op            23 B/op          1 allocs/op
BenchmarkCheckingAndCalling-4           20000000                67.5 ns/op            23 B/op          1 allocs/op
BenchmarkCheckingAndNotCalling-4        20000000                65.8 ns/op            23 B/op          1 allocs/op
```

Nil handling would be a marginal speedup in default cases, however if you change the value from nil to something else you'll pay the price of that if that does the checking (again it's marginal so it shouldn't bother no one).

I don't know how to judge this really. However having a nil value as a default may be helpful also because user can't shoot himself in the foot now. In the original design if you assign nil to `ErrorMarshalFunc` it would panic on invocation. 

Thoughts?